app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -142,7 +142,11 @@ def save_security_groups_inventory(ems, hashes, target = nil)
     firewall_rules       = ems.security_groups.collect(&:firewall_rules).flatten
     firewall_rules.each do |fr|
       fr_hash = firewall_rule_hashes[fr.id] || {}
-      fr_hash[:source_security_group_id] = fr_hash.fetch_path(:source_security_group, :id)
+      unless fr_hash[:source_security_group].nil?
```

Comment:
hm, this code is shared, so we should be be switching this, I think you can really fix this up in the particular parser, where there fr_hash[:security_group] needs to be a hash containing the security_group, not just the ems_ref.